src/corehost/common/pal.unix.cpp

```diff
@@ -112,11 +124,64 @@ bool pal::is_path_rooted(const pal::string_t& path)
     return path.front() == '/';
 }
 
+bool pal::get_default_breadcrumb_store(string_t* recv)
+{
+    recv->clear();
+    pal::string_t ext;
+    if (!get_default_extensions_directory(&ext))
+    {
+        trace::info(_X("Breadcrumb store not obtained because dotnet extensions could not be obtained"));
+        return false;
+    }
+
+    append_path(&ext, _X("breadcrumbs"));
+    if (!pal::directory_exists(ext))
+    {
+        trace::info(_X("Breadcrumb store does not exist in [%s]"), ext.c_str());
+        return false;
+    }
+
+    if (access(ext.c_str(), (R_OK | W_OK)) != 0)
+    {
+        trace::info(_X("Breadcrumb store [%s] is not ACL-ed with rw-"), ext.c_str());
+        return false;
+    }
+
+    recv->assign(ext);
+    return true;
+}
+
 bool pal::get_default_extensions_directory(string_t* recv)
 {
     recv->clear();
-    append_path(recv, _X("opt"));
-    append_path(recv, _X("dotnetextensions"));
+    pal::string_t ext;
+    if (pal::getenv(_X("DOTNET_EXTENSIONS"), &ext) && pal::realpath(&ext))
+    {
+        // We should have the path in ext.
+        trace::info(_X("Realpath $DOTNET_EXTENSIONS [%s]"), ext.c_str());
+    }
+    
+    if (!pal::directory_exists(ext))
```

Comment:
It will be zero inited by the default constructor and directory exists will early out on an empty string.